### DISCUSSION TITLE
How to make floating components draggable?

### DISCUSSION THREAD

**Discussion by lukasbash** [mantinedev/mantine#5564] [Q&A]
I want to apply some dragging mechanism to any kind of floating elements like modals or popovers. Do you by any chance have an example how to achieve this? I was thinking about the `use-move` hook in combination with that but for the modals manager there is no root render callback I could use.

Any feedback? Or do you maybe see this as a doable feature request?

**Comment by rtivital** [comment]
Mantine does not provide this functionality neither in hooks nor in components. use-move hook is not intended to be used this way. I suggest using a third-party library for that or implementing it on your side.

**Reply by lukasbash** [reply]
Well I am not against using a third party API for this, but in order to do this I need some kind of entry point where mantine renders the most upper component of my modal. Considering I want to use for example [react-draggable](https://github.com/react-grid-layout/react-draggable), I need some callback to wrap the mantine component like:
```jsx
// Having a prop like this on the modals provider would be awesome.
// Either a default renderer is passed as well, or the actual component will be exposed.
onRenderModal={(props, defaultRender) => {
  return <Draggable>{defaultRender(props)}</Draggable>;
}}
```

At the current state is it not possible for me to integrate a third party API into the context modal concept of mantine.

Can we somehow get this as a feature request maybe?

**Reply by rtivital** [reply]
https://mantine.dev/core/modal/#compound-components
